module/programs/helper.js

```diff
@@ -1081,14 +1093,17 @@ module.exports = class ProgramsHelper {
               status: httpStatusCode.bad_request.status
           }
         }
-        console.log("kafka push status :",pushProgramUsersDetailsToKafka)
+        
         let joinProgramDetails = joinProgram.toObject();
-  
+        // if programUsersDetails[0].consentShared === true which means the data is already pushed to Kafka once, So revert pushProgramUsersDetailsToKafka to false
+        if (programUsersDetails.length > 0 && programUsersDetails[0].hasOwnProperty('consentShared') && programUsersDetails[0].consentShared === true) {
+          pushProgramUsersDetailsToKafka = false;
```

Comment:
@VISHNUDAS-tunerlabs if again and again I call join API with consentShared = true what will happen?

Reply:
@aks30 First time you call with consent shared is true. That time only kafka event will be triggered.

Reply:
@VISHNUDAS-tunerlabs why do we have 2 code blocks defined for determining the value of pushProgramUsersDetailsToKafka, Line 1060-1065 and 1095-1098? Can we not have just one?

Reply:
@aks30 Both conditions serve different purposes, and merging them would lead to incorrect logic. so we can’t have single if condition, But we can make it into a single code block, that change is added